### DISCUSSION TITLE
How do you correctly arrange tiles?

### DISCUSSION THREAD

**Discussion by TwinTailzz** [TrianglyRU/OrbinautFramework#35] [Q&A] [answered]
Hi! im currently trying to use the sonic 3 collision tileset and i was simply wondering if there was some sort of breakdown of what tiles specifically belong together?

Sorry in advance if i missed something in the documentation about it but im trying to wrap my head around it.

My main questions are:

> What other tiles can go together other than the immediately obvious ones?

> Can tiles be used in any formation as long as they create a clean slope?

> I noticed that there were tiles that create an "unfinished" slope, meaning i couldn't find any tiles that cleanly line up with it, but an awkward arrangement of tiles would get it connected to ground, does every tile have a "neat" way to integrate them? or is it for situational / unique terrain?

I dont have anything in particular im working on that i can show, just trying to parse out how intricate i can get with the tiles.

**Comment by N8rtot** [comment, marked as answer]
Pretty much any configuration of collision works for whatever you may need, but heres what id say
1. Any collision tiles can work together, just make sure not to rotate them as they will break; Flipping horizontally and vertically is safe
2. Yup, you can create a slope out of any configuration of collision tiles.
3. If im correct, the extra collision sheets (i.e, s1, s2, s3) should all line up to the tileart from those respective games zones. So alot of wierd looking collision tiles in those may belong to a certain slope in a certain level. If you go to the Test Stage, you can see an example of how the collision tiles line up with the custom made slopes. (Keep in mind this doesnt mean they can ONLY be used for that situational spot)

A few key things to remember too -
1. Collision tiles tend to be organized by slope, and are broken down and laid out left to right. Although keep in mind that slopes will also borrow collision tiles from other slopes, so you may be working on a slope and have it nearly completed, but have to find the last piece as its in a different spot due to another slope having the piece first.
-Heres a video example if it helps

https://github.com/user-attachments/assets/1990d1aa-7295-4447-8d48-8c0cc056fdb6

2. For grassy ground, or similar spots where there may be art hiding the flat ground that the characters run on, you dont need to find collision that perfectly lines up, infact most grassy levels from the classic games have pretty bumpy and uneven collision! 

3. SonLVL is an amazing tool to use if you're trying to replicate an existing zone and see how the collision is set up and works there too. While im at it, shoutout to the website Zone0 for being an equally amazing tool for studying level design but thats besides the point. 

4. You can always create your own collision sheet, so if you want any custom slopes, hills, or the like, you can absolutely get crazy with it. 
If you want to do so, make sure to check out the wiki for further, and more indepth detail, [you can just click this text to go there](https://github.com/TrianglyRU/OrbinautFramework/wiki/7.-Creating-a-New-Collision-Tileset)

-Also known program that can often be helpful with creating custom slopes (and in this case, custom collision for said slopes) would be "Sidescroller Tile Set Maker" by MrLevRocks on itch.io

**Reply by TwinTailzz** [reply]
Thanks for the reply! Yeah, so i did notice that most slopes were grouped together but my suspicion of other slope shapes using tiles from a different slope shape is now confirmed. 

I was worried i was building out slopes wrong because i wasn't using the tiles grouped together.

it is also good to know that those weird tiles i was talking about correspond to areas in specific zones, so now i know.

Also i will definitely check out those tools you mentioned so thanks for the pointers!

i decided to make a brush with all the slope combinations and orientations so far.

Thanks again!

![image_2024-09-11_160402567](https://github.com/user-attachments/assets/f54e4ec7-7834-4264-bf3b-34bf415e558a)

**Reply by TrianglyRU** [reply]
I should also mention that you don't really need a pre-rotated variation of each tile, especially flipped vertically, as the game may not behave with them the way you'd expect to. Making a tile going from left to right is pretty much enough. Having more tiles meaning you'll have to create a longer angle data array which is time consuming

**Reply by astroblema** [reply]
Sorry for necrobumping, but how would i go about using the brush Twintailzz created? Or how can I build it on my own?

**Reply by TrianglyRU** [reply]
https://www.youtube.com/watch?v=apzyZ4SBTdg

**Comment by TrianglyRU** [comment]
This discussion is now referenced in the wiki
